packages/gasket-plugin-https-proxy/README.md

```diff
@@ -0,0 +1,101 @@
+# @gasket/plugin-https-proxy
+
+Create an HTTPS proxy server with Gasket to use as a sidecar for Next.js apps 
+which do not handle HTTPS. This can be useful for local development when you 
+want to test HTTPS features and use secure cookies for authentication, etc.
+It can also be used in production if it is necessary for your application to
+handle HTTPS requests on the container.
+
+## Installation
+
+```
+npm i @gasket/plugin-https-proxy
+```
+
+Update your `gasket` file plugin configuration:
+
+```diff
+// gasket.js
+
++ import pluginHttpsProxy from '@gasket/plugin-https-proxy';
+
+export default makeGasket({
+  plugins: [
++   pluginHttpsProxy
+  ]
+});
+```
+
+## Configuration
+
+To be set under `httpsProxy` in the `gasket.js` file.
+
+This uses the [http-proxy] package and its full set of available [options].
+
+the `target` is 
+required to
+
+```diff
+// gasket.js
+export default makeGasket({
++  httpsProxy: {
++    protocol: 'https',
++    hostname: 'my-host.com',
++    port: 443,
++    xfwd: true,
++    ws: true,
++    target: {
++      host: 'localhost',
++      port: 80
++    }
++  }
+});
+```
+
+The above example forwards https requests on port `443` to `localhost:80`.
+
+The `protocol` and `hostname` are only used for logging about the proxy server
```

Comment:
I wonder if nesting things under a `source` property to complement `target` would make that more intuitive.

Reply:
Also, one thing that's a bit strange is that this is called an https proxy specifically but you have to specify the protocol as `https` as well. Should this be a general purpose proxy plugin instead?

Reply:
Yes it is mostly pass-through configuration. I could default the protocol to `https` - I'll make that change. So, technically yea, the underlying `http-proxy` package could be used to proxy to other services or something, but I don't see it being useful that way in a Gasket context. I went back and forth on the name, but an HTTPS proxy is the specific gap intended to be filled with this plugin.